### PR TITLE
IBP-5519 Added DTOs and services for getting the Germplasm Lists in BRAPI Context

### PR DESCRIPTION
**Pull request by mderamos**

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/api/brapi/GermplasmListServiceBrapiImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+@Service
+@Transactional
+public class GermplasmListServiceBrapiImpl implements GermplasmListServiceBrapi {
+
+	@Autowired
+	private UserService userService;
+
+	private final DaoFactory daoFactory;
+	private final HibernateSessionProvider sessionProvider;
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/api/brapi/GermplasmListServiceBrapiImpl.java:30]: Remove unused `sessionProvider`

File: src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java [hunk 2]
```diff
@@ -883,4 +903,118 @@ public long countSearchGermplasmList(final GermplasmListSearchRequest germplasmL
...
+
+	public long countGermplasmListDTOs(final GermplasmListSearchRequestDTO requestDTO) {
+		final SQLQuery sqlQuery = this.getSession().createSQLQuery(this.createCountListsQueryString(requestDTO));
+		this.addListSearchParameters(sqlQuery, requestDTO);
+		return ((BigInteger) sqlQuery.uniqueResult()).longValue();
+	}
+
+	public List<GermplasmListDTO> searchGermplasmListDTOs(final GermplasmListSearchRequestDTO requestDTO, final Pageable pageable) {
+		final SQLQuery sqlQuery = this.getSession().createSQLQuery(this.createListsQuery(requestDTO));
+		if (pageable != null) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java:915]: This code is duplicated:
`		if (pageable != null) {
			sqlQuery.setFirstResult(pageable.getPageSize() * pageable.getPageNumber());
			sqlQuery.setMaxResults(pageable.getPageSize());
		}`

Comment (darla-leafnode) [reply src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java:915]: You can use `addPaginationToSQLQuery` instead

File: src/main/java/org/generationcp/middleware/domain/search_request/brapi/v2/GermplasmListSearchRequestDTO.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	private String listType;
+	private String listName;
+	private String listDbId;
+	private String listSource;
+	private String externalReferenceID;
+	private String externalReferenceSource;
+
+	public GermplasmListSearchRequestDTO(final String listType, final String listName, final String listDbId, final String listSource,
+		final String externalReferenceID, final String externalReferenceSource) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/domain/search_request/brapi/v2/GermplasmListSearchRequestDTO.java:18]: `listSource` parameter is unused

File: src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	@Test
+	public void testCountGermplasmListDTOs() {
+		final GermplasmListSearchRequestDTO requestDTO = new GermplasmListSearchRequestDTO();
+		requestDTO.setListDbId(this.germplasmListId.toString());
+		Assert.assertEquals(1, (int) this.germplasmListServiceBrapi.countGermplasmListDTOs(requestDTO));
+	}
+
+	@Test
+	public void testSearchGermplasmListDTOs() {
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java:58]: Add more tests for filtering scenarios

**Review by darla-leafnode — CHANGES_REQUESTED** [review]

File: src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java (hunk 4 above)
Comment (mderamos) [reply src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java:58]: Hi @darla-leafnode , only the following filters are used in the method being tested: listName, listDbId, externalReferenceID, externalReferenceSource. 

listSource is not supported. As for listType, we're only accepting "germplasm" value which then returns all the germplasm lists regardless of the "listtype" value in the database.

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**